packages/cashier/src/pages/payment-agent/payment-agent-card/paymen-agent-card-deposit-details.jsx

```diff
@@ -0,0 +1,131 @@
+        </div>
+    );
+};
+
+Detail.propTypes = {
+    action: PropTypes.string,
+    children: PropTypes.oneOfType([PropTypes.array, PropTypes.element, PropTypes.string]),
+    className: PropTypes.string,
+    has_red_color: PropTypes.bool,
+    icon: PropTypes.string,
+    title: PropTypes.string,
+};
+
+const PaymentAgentCardDetails = ({ payment_agent }) => {
+    const payment_agent_phones = toJS(payment_agent.phones);
+    const detail_titles = {
+        email: localize('Email'),
+        phone_number: localize('Phone number'),
+        deposit_commission: localize('Commission on deposits'),
+        withdrawal_commission: localize('Commission on withdrawal'),
+        min_withdrawal: localize('Minimum withdrawal'),
+        max_withdrawal: localize('Maximum withdrawal'),
+    };
+
+    return (
+        <div className='payment-agent-card__deposit-details-container'>
+            {payment_agent_phones && (
+                <Detail action='tel' icon='Phone' title={detail_titles.phone_number}>
+                    {Array.isArray(payment_agent.phones)
+                        ? payment_agent.phones.map(phone => phone.phone_number)
+                        : payment_agent.phones}
+                </Detail>
+            )}
+            {payment_agent.email && (
+                <Detail
+                    action='mailto'
+                    icon='Email'
+                    rel='noopener noreferrer'
+                    target='_blank'
+                    title={detail_titles.email}
+                    has_red_color
+                >
+                    {payment_agent.email}
+                </Detail>
+            )}
+            {payment_agent.min_withdrawal && (
+                <Detail icon='MinimumWithdrawal' title={detail_titles.min_withdrawal}>
+                    <Money amount={payment_agent.min_withdrawal} currency={payment_agent.currency} show_currency />
+                </Detail>
+            )}
+            {payment_agent.deposit_commission && (
+                <Detail
+                    icon='CommissionDeposit'
+                    className='deposit-commission'
+                    title={detail_titles.deposit_commission}
+                >
+                    {`${payment_agent.deposit_commission}%`}
+                </Detail>
+            )}
+            {payment_agent.max_withdrawal && (
+                <Detail icon='MaximumWithdrawal' title={detail_titles.max_withdrawal}>
+                    <Money amount={payment_agent.max_withdrawal} currency={payment_agent.currency} show_currency />
+                </Detail>
+            )}
+            {payment_agent.withdrawal_commission && (
+                <Detail
+                    icon='CommissionWithdrawal'
+                    className='withdrawal_commission'
+                    title={detail_titles.withdrawal_commission}
+                >
+                    {`${payment_agent.withdrawal_commission}%`}
+                </Detail>
+            )}
+        </div>
```

Comment:
```suggestion
        <div className='payment-agent-card__deposit-details-container'>
            {payment_agent_phones && <PaymentAgentPhonesDetails />}
            {payment_agent.email && <EmailDetails />}
            {payment_agent.min_withdrawal && <MinWithdrawalDetails />}
            {payment_agent.deposit_commission && <DepositCommissionDetails />}
            {payment_agent.max_withdrawal && <MaxWithdrawalDetails />}
            {payment_agent.withdrawal_commission && <WithdrawalCommissionDetails />}
        </div>
```
@george-usynin-binary I would suggest extracting them into helper functions for better readability.